### PR TITLE
Use the stacktrace to determine function origin

### PR DESCRIPTION
**Pull request by bahlo**
Middleware is called from `worker.js` in Vercel and from
`evalmachine.<anonymous>` in local development.

### CODE REVIEW

**Comment by schehata** [comment]
tested locally, works as expected 👍

**Review by schehata — APPROVED** [review]

File: src/withAxiom.ts [hunk 1]
```diff
@@ -122,10 +122,10 @@ function isNextConfig(param: WithAxiomParam): param is NextConfig {
...
 
 function isApiHandler(param: WithAxiomParam): param is NextApiHandler {
-  // This is pretty hacky, but if you call withAxiom in a serverless function,
-  // the environment variable will be set.
-  // The middleware runs on CloudFlare workers which doesn't expose that env.
-  return typeof param == 'function' && !!process.env.LAMBDA_TASK_ROOT;
+  // Middleware is called from `worker.js` in Vercel and from
+  // `evalmachine.<anonymous>` in local development.
+  const caller = getCaller();
+  return typeof param == 'function' && caller != 'worker.js' && caller != 'evalmachine.<anonymous>';
```
Comment (bahlo) [inline src/withAxiom.ts:128]: We need to check the `LAMBDA_TASK_ROOT` env if it‘s worker.js to detect api functions that are called worker.js on Vercel

**Comment by schehata** [comment]
works locally, testing now with edge functions.

**Comment by bahlo** [comment]
Closing for #23

**Pull request closed without merging**